ekm/signer_key_manager_test.go

```diff
@@ -788,3 +790,133 @@ func TestEkmListAccounts(t *testing.T) {
 	require.NoError(t, err)
 	require.Equal(t, 2, len(accounts))
 }
+
+func TestConcurrentSlashingProtectionAttData(t *testing.T) {
+	require.NoError(t, bls.Init(bls.BLS12_381))
+
+	km := testKeyManager(t, nil)
+
+	sk1 := &bls.SecretKey{}
+	require.NoError(t, sk1.SetHexString(sk1Str))
+	require.NoError(t, km.AddShare(sk1))
+
+	currentSlot := km.(*ethKeyManagerSigner).storage.Network().EstimatedCurrentSlot()
+	currentEpoch := km.(*ethKeyManagerSigner).storage.Network().EstimatedEpochAtSlot(currentSlot)
+
+	highestTarget := currentEpoch + minSPAttestationEpochGap + 1
+	highestSource := highestTarget - 1
+
+	attestationData := &phase0.AttestationData{
+		Slot:            currentSlot,
+		Index:           1,
+		BeaconBlockRoot: [32]byte{1, 2, 3, 4, 5, 6, 1, 2, 3, 4, 5, 6, 1, 2, 3, 4, 5, 6, 1, 2, 3, 4, 5, 6, 1, 2, 3, 4, 5, 6, 1, 2},
+		Source: &phase0.Checkpoint{
+			Epoch: highestSource,
+			Root:  [32]byte{},
+		},
+		Target: &phase0.Checkpoint{
+			Epoch: highestTarget,
+			Root:  [32]byte{},
+		},
+	}
+
+	// Define function to concurrently attempt signing.
+	signAttestation := func(wg *sync.WaitGroup, errChan chan error) {
+		defer wg.Done()
+		_, _, err := km.(*ethKeyManagerSigner).SignBeaconObject(
+			attestationData,
+			phase0.Domain{},
+			sk1.GetPublicKey().Serialize(),
+			spectypes.DomainAttester,
+		)
```

Comment:
we can potentially add a check for the sig here

to get a sig to hardcode, we can run this test once without parallelism just to print the sigs we know are good

Reply:
To achieve this, we’d need to adjust the tests to use a mock network to obtain a static slot for signing (or use some MaxUint64) value, which would ensure consistent signature values. This change would require updating all related tests to use the new mock network. Do we still want to proceed with this approach in the current PR?

Reply:
added sig verification instead